File: common-res/src/main/java/com/yyxnb/common_res/service/impl/LoginImpl.java
package com.yyxnb.common_res.service.impl;

import android.content.Context;

import com.alibaba.android.arouter.facade.annotation.Autowired;
import com.alibaba.android.arouter.launcher.ARouter;
import com.yyxnb.common_res.service.LoginService;
import com.yyxnb.common_res.constants.LoginRouterPath;
import com.yyxnb.what.arch.base.IFragment;

/**
 * ================================================
 * 作    者：yyx
 * 日    期：2021/03/16
 * 描    述：对LoginService包装，业务方直接调用，无需再自己初始化service类
 * ================================================
 */
public class LoginImpl {

    @Autowired(name = LoginRouterPath.SERVICE)
    protected LoginService service;

    private static volatile LoginImpl mInstance = null;

    private LoginImpl() {
        //初始化
        ARouter.getInstance().inject(this);
    }

    public static LoginImpl getInstance() {
        if (null == mInstance) {
            synchronized (LoginImpl.class) {
                if (null == mInstance) {
                    mInstance = new LoginImpl();
                }
            }
        }
        return mInstance;
    }

    public void start(Context context) {
        service.start(context);
    }

    public IFragment mainPage(Context context) {
        return service.mainPage(context);
    }

    public IFragment showPage(Context context) {
        return service.showPage(context);
    }

    /**
     * 是否登录
     */
    public boolean isLogin() {
        return service.isLogin();
    }

    /**
     * 退出
     */
    public void loginOut() {
        service.loginOut();
    }

}

File: common_base/src/main/java/com/yyxnb/common_base/arouter/service/impl/LoginImpl.kt
package com.yyxnb.common_base.arouter.service.impl;

import android.content.Context;

import com.alibaba.android.arouter.facade.annotation.Autowired;
import com.alibaba.android.arouter.launcher.ARouter;
import com.yyxnb.common_base.arouter.service.LoginService;
import com.yyxnb.common_base.bean.UserBean;

import static com.yyxnb.common_base.arouter.ARouterConstant.LOGIN_SERVICE;

/**
 * 对LoginService包装，业务方直接调用，无需再自己初始化service类
 */
public class LoginImpl {

    @Autowired(name = LOGIN_SERVICE)
    protected LoginService mLoginService;

    private static LoginImpl mLoginImpl = null;

    public static LoginImpl getInstance() {
        if (mLoginImpl == null) {
            synchronized (LoginImpl.class) {
                if (mLoginImpl == null) {
                    mLoginImpl = new LoginImpl();
                }
                return mLoginImpl;
            }
        }
        return mLoginImpl;
    }

    private LoginImpl() {
        //初始化
        ARouter.getInstance().inject(this);
    }

    /**
     * 跳转登录 ，建议直接路由跳转
     */
    public void login(Context context) {
        mLoginService.login(context);
    }

    /**
     * 是否登录
     */
    public boolean isLogin() {
        return mLoginService.isLogin();
    }

    /**
     * 退出
     */
    public void loginOut() {
        mLoginService.loginOut();
    }

    /**
     * 获取用户信息
     */
    public UserBean getUserInfo() {
        return mLoginService.getUserInfo();
    }

    /**
     * 更新用户信息
     */
    public void updateUserInfo(UserBean userBean) {
        mLoginService.updateUserInfo(userBean);
    }

}
